processor/metricsgenerationprocessor/utils.go

```diff
@@ -126,15 +231,25 @@ func generateMetric(from pmetric.Metric, operand2 float64, operation string, log
 	return to
 }
 
-// Append the scope metrics with the new metric
-func appendMetric(ilm pmetric.ScopeMetrics, newMetric pmetric.Metric, name, unit string) pmetric.Metric {
-	metric := ilm.Metrics().AppendEmpty()
-	newMetric.MoveTo(metric)
+// Append the new metric to the scope metrics. This will only append the new metric if it
+// has data points.
+func appendNewMetric(ilm pmetric.ScopeMetrics, newMetric pmetric.Metric, name, unit string) {
```

Comment:
The logic of only adding new metrics if they have valid data points was going to be copied in multiple places, so I decided to move it here to a centralized location.